cpp/foxglove/src/server.cpp

```diff
@@ -239,21 +241,52 @@ FoxgloveResult<WebSocketServer> WebSocketServer::create(
     };
   }
 
+  if (options.sink_channel_filter) {
+    sink_channel_filter = std::make_unique<SinkChannelFilterFn>(options.sink_channel_filter);
+
+    c_options.sink_channel_filter_context = sink_channel_filter.get();
+    c_options.sink_channel_filter =
+      [](const void* context, const struct foxglove_channel_descriptor* channel) -> bool {
+      try {
+        if (!context) {
+          return true;  // Default to allowing if no filter
+        }
+        auto* filter_func = static_cast<const SinkChannelFilterFn*>(context);
+        auto metadata = buildMetadata_(channel->metadata);
```

Comment:
Similarly wondering if there should be a static class method `ChannelDescriptor::from_raw(foxglove_channel_descriptor*)` or custom constructor that wraps this so it's not being duplicated in both the mcap and server implementation